### DISCUSSION TITLE
Mojo support

### DISCUSSION THREAD

**Discussion by martin-braun** [beeware/briefcase#1434] [Ideas]
Hi. I'm a big fan of type safe code that runs reasonable fast, that's why I always try to pick something native and would pick TS over JS if I need JS. That's also why I never consider Python for larger projects, although I think Python has by far the best syntax.

However, things are going to change with [Mojo](https://github.com/modularml/mojo). Here is a good introduction video which is short and to the point: https://www.youtube.com/watch?v=V4gGJ7XXlC0

For a long while I'm also looking for a tech stack that is able to target all platforms without any WebView or Canvas trickery. I want a DOM on web targets, I want true native UI on mobile and desktop apps. And by all means, I don't want things like React, otherwise I would probably stick with React Native.

BeeWare seems to check all marks if it would run on a more performant interpreter that has better performance and type safety. Mojo seem to bring that opportunity. There will be interpreters for desktops, although I don't know if there ever will be any for mobile.

I know there is probably a long way to go. After Mojo is not even publicly released yet. APIs will change, stuff will be unstable or break. There is a long road ahead.

I'm opening this discussion to ask for first impressions by the team behind BeeWare. It's certainly to early to make a decision on that, but I hope it will still bring some interest to you, I hope you are as excited as I am about Mojo and what it can bring to the table.

**Comment by freakboy3742** [comment]
Honestly - I have no idea what to make of Mojo at this point. 

A well funded startup has announced a new programming language that is "like Python but faster and typed". It's not open sourced (but they promise it will be! Super pinky swear!), and there's no way to try it out beyond a sandbox. But it's totally going to solve *all* your problems!

At *best*, I'm sceptical. 

Projects that are funded by a single VC-backed company have a nasty habit of disappearing once the company behind them runs out of cash (witness the recent mess surrounding Gatsby and Terraform, to pick 2 examples). Unless Mojo is *actually* open sourced, *and* able to develop a significant developer *and maintainer* community *outside* of Modular as a company - and fast - it's hard to see how it will survive. 

However, assuming Mojo materializes in a meaningful way - what will BeeWare's relationship with Mojo be? It's literally impossible to tell at this point in time. They appear to be targeting the Python community with their messaging - but they've also been pretty explicit that Mojo *isn't* a language superset, so the migration or compatibility path for Python tools into the Mojo ecosystem is *completely* opaque. 

*If* it turns out there's a way for Mojo to use BeeWare tooling, or to be compatible with BeeWare tooling, then I'm not fundamentally opposed to supporting that use case. However, as with most suggestions regarding "edge use cases", my preference is for any support to start life as an externally managed plugin, rather than as direct features in the core tool. If it turns out that a BeeWare tool can't support a required mode of extension, adding plugin hooks to allow extension is more likely than merging explicit support for a third-party tool.

**Reply by martin-braun** [reply]
This is reasonable. Let's be sceptical indeed, that's probably the healthier stance. They claim there will be the first version to download in a couple of weeks. I will check it out, notify you, let's see where this goes and if its worth to consider after all. If their claims are true, you can boost performance over 10x by simply swapping the interpreter on desktop systems, but I would definitely wait for a stable open-source release before even considering it. I'm still hopeful, as the creator is the same one who made the Swift language and the LLVM toolchain.

Lastly, thanks for the input on BeeWare with Mojo. That's indeed helpful to me. As I learned BeeWare is really modular, so it makes sense to introduce a plugin when it really comes to that point. The hard part though is that a new language requires plugins for all targets, or code re-use is impossible across platforms.
